src/connection-pool-party.js

```diff
@@ -194,12 +212,15 @@ export default class ConnectionPoolParty extends EventEmitter {
   close = cb => Promise.all(
     this.pools.map(pool => pool.connection.close()),
   )
-    .then(() => {
-      this.pools = [];
-    })
     .catch((err) => {
       this.emit('error', err);
+    })
+    .then(() => {
```

Comment:
Why delaying the cleanup after `emit`-ing the `err`?  Seems like it is better to `cleanup` before `emit`.

Reply:
was just trying to reduce duplication, since i had to do the cleanup in a `then` before the `catch` and also in the `catch`.  do you see any issue with this shortcut?

Reply:
I'm just concerned about the consumer taking action on the `err` event and operating on the pool before we finish cleaning things up.

Reply:
i suppose that's a possibility, i'll revert